be/src/storage/schema_change.cpp

```diff
@@ -853,6 +854,11 @@ RowBlockAllocator::~RowBlockAllocator() {
     }
 }
 
+bool RowBlockAllocator::is_memory_enough_to_sort(size_t num_rows, size_t allocated_rows) {
+    size_t row_block_size = _row_len * (num_rows - allocated_rows);
```

Comment:
Is num_rows greater than allocated_rows in any case?
IMO, num_rows may be less than allocated_rows in most cases

Reply:
allocated_rows is max num_rows of historical row-block, but it this does not guarantee that allocated_rows is always greater than num_rows because we may filter some rows in `next_block`, so when num_rows is greater than allocated_rows, we need more memory to do block-sort.  If we don't check memory is enough to do block-sort or not, block-sort may return failure which result in schema-change failure

Reply:
size_t is unsigned, if num_rows is less than allocated_rows, it will return an invalid number. So I think ssize_t of int64_t is a better choice.

Reply:
add a check between num_rows and allocated_rows instead of using ssize_t to avoid  implicit conversion from signed to unsigned. 
see commit https://github.com/StarRocks/starrocks/pull/225/commits/8f7ed607fd7c173926d15bbbc2188b22e8717f18